_episodes/02-shell-basics.md

```diff
@@ -1,31 +1,418 @@
+
+A related idea is the *type* of a file.
+Broadly speaking, files are of two types:
+
+1. **Text files** are files that contain data in human-readable
+format such as letters and numbers.
+A text file can be easily viewed and edited using any text-editor,
+such as Notepad, TextEdit or Atom.
+
+2. **Binary files** are files that contain data in machine-readable format.
+Special programs are required to open, view or edit binary files.
+For example, a `.docx` file can only be understood by programs
+like Microsoft Word or OpenOffice Writer.
+
+> ## Text v/s binary files
+>
+> The commands `cat`, `head` and `tail` are useful
+> for viewing files.
+>
+> Command           | Action
+> ------------------|--------------------------
+> `cat FILE`        | Print the contents of the file `FILE`
+> `cat FILE1 FILE2` | Concatenate (join) the files `FILE1` and `FILE2` and print the result
+> `head FILE`       | Print the first 10 lines of the file `FILE`
+> `head -n 4 FILE`  | Print the first 4 lines of the file `FILE`
+> `tail FILE`       | Print the last 10 lines of the file `FILE`
+> `tail -n 4 FILE`  | Print the last 4 lines of the file `FILE`
+>
+> Print the first few lines of the two files
+> `notes.txt` and `solar.pdf`.
+> Can you explain the output of each command?
+>
+{: .challenge}
+
+As a user of the cluster,
+you will frequently be working with text files
+such as scripts, source code, documentation, configuration files, etc.,
+so it's important that you learn to view and edit these files efficiently.
+
+Graphical text editors like Notepad or Atom are unavailable on the cluster.
+Instead, you must use a terminal-based text editor
+to edit text files.
+`nano` is a very simple editor that can be used
+for basic text editing:
+
+~~~
+[nelle@login001 shell-basics]$ nano notes.txt
+~~~
+{: .bash}
+
+When you open a text file such as `notes.txt` using `nano`,
+you should see the following screen:
+
+~~~
+  GNU nano 2.3.1		File: notes.txt
+
+- finish experiments
+- write thesis
+- get post-doc position (pref. with Dr. Horrible)
+
+
+
+
+
+
+
+
+
+
+
+^G Get Help  ^O WriteOut  ^R Read File ^Y Prev Page ^K Cut Text  ^C Cur Pos
+^X Exit      ^J Justify   ^W Where Is  ^V Next Page ^U UnCut Text^T To Spell
+~~~
+{: .output}
+
+Here, you can add or remove text.
+Along the bottom of the editor window,
+you will see that you can type `^O` (or `Control+O`)
+to write out or "save" any changes you make,
+and `Control+X` to exit `nano`.
+When prompted for "Yes" or "No",
+you can type the `Y` or `N` key respectively.
+
+> ## Other text editors
+> While `nano` is well-suited for basic-editing
+> and a good choice for beginners,
+> you will probably want to learn a text editor designed for
+> [real programmers](https://xkcd.com/378/)
+> such as [`vim`](https://en.wikipedia.org/wiki/Vim_(text_editor))
+> or [`emacs`](https://en.wikipedia.org/wiki/Emacs).
+> Type `vimtutor` on the command-line for a quick introduction
+> to `vim`.
```

Comment:
Maybe add: "For the sake of staying general, we will use `nano` throughout this workshop as a reference."